src/game/SpellEffects.cpp

```diff
@@ -7147,10 +7155,20 @@ void Spell::EffectSummonDeadPet(SpellEffectIndex /*eff_idx*/)
 {
     if (m_caster->GetTypeId() != TYPEID_PLAYER)
         return;
+
+    bool wasDespawned = false;
     Player* _player = (Player*)m_caster;
     Pet* pet = _player->GetPet();
     if (!pet)
-        return;
+    {
+        pet = new Pet;
+        if (!pet->LoadPetFromDB(_player, 0, 0, false, true))
```

Comment:
well, if those initial checks are being done in checkcast and this is just a failsafe then I'm all for it, but shouldn't it output an error message as well, then?